### PR TITLE
feat: Tasklist/persist task competition to tasklist-listview

### LINKED ISSUE #21845: [Root Task] Search Task by Variable
```[tasklist]
### Tasks
- [ ] https://github.com/camunda/camunda/issues/21843
- [ ] https://github.com/camunda/camunda/issues/21846
- [ ] https://github.com/camunda/camunda/issues/21848
- [ ] https://github.com/camunda/camunda/issues/21849
```

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

- Added constructors and methods to `ListViewJoinRelation` and `VariableListViewEntity` for enhanced initialization.
- Defined and implemented the `ListViewStore` interface for variable manipulation in Elasticsearch and OpenSearch.
- Integrated variable persistence in the list view upon task completion and ensured job worker variables are not persisted.
- Updated `TaskService` and `VariableService` to handle new variable persistence logic.
- Extended **integration** and unit tests to cover new functionality and changes.

## Requirements

- Persist Task Variables (Not Draft Variables) on User Tasks Completition on tasklist-list-view indice. This means output variables from Tasks will become Process Variables, also the process variables on the moment of task completition will be persisted as Task Variables
- Remove Job Workers Task variables - As it is not possible verify when a variable come from V1 and V2, to avoid have documents that wont be able related to any parent (Once we have no Job Workers), On the task competition of JOB WORKERS, remove the Task Variables. But Process Variables will be persisted normally. 

#closes #21876

### CODE REVIEW

**Comment by nathansandi** [comment]
Preparing PR description...

**Comment by nathansandi** [comment]
**[PR Description](https://github.com/camunda/camunda/pull/21888)** updated to latest commit (https://github.com/camunda/camunda/commit/f49ce79ef5b4a44cab8b007a9a3f2181c7b5beed)

File: tasklist/els-schema/src/main/java/io/camunda/tasklist/store/elasticsearch/ListViewStoreElasticSearch.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+              .source(
+                  SearchSourceBuilder.searchSource()
+                      .query(
+                          termQuery(
+                              TasklistListViewTemplate.VARIABLE_SCOPE_KEY, flowNodeInstanceId)));
+
+      final var searchResponse = esClient.search(searchRequest, RequestOptions.DEFAULT);
+      final SearchHit[] hits = searchResponse.getHits().getHits();
+
+      for (final SearchHit hit : hits) {
```
Comment (houssain-barouni) [inline tasklist/els-schema/src/main/java/io/camunda/tasklist/store/elasticsearch/ListViewStoreElasticSearch.java:72]: I think you need to scroll, otherwise you will delete only the first 10 hits

File: tasklist/els-schema/src/main/java/io/camunda/tasklist/store/elasticsearch/ListViewStoreElasticSearch.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  @Override
+  public List<VariableListViewEntity> getVariablesByVariableName(final String varName) {
+    try {
+      final SearchRequest searchRequest =
+          ElasticsearchUtil.createSearchRequest(tasklistListViewTemplate)
+              .source(
+                  SearchSourceBuilder.searchSource()
+                      .query(termQuery(TasklistListViewTemplate.VARIABLE_NAME, varName)));
+
+      final var searchResponse = esClient.search(searchRequest, RequestOptions.DEFAULT);
```
Comment (houssain-barouni) [inline tasklist/els-schema/src/main/java/io/camunda/tasklist/store/elasticsearch/ListViewStoreElasticSearch.java:102]: same thing, you may need to do scrolling here?

File: tasklist/els-schema/src/main/java/io/camunda/tasklist/store/elasticsearch/ListViewStoreElasticSearch.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+          .map(
+              hit ->
+                  fromSearchHit(
+                      hit.getSourceAsString(), objectMapper, VariableListViewEntity.class))
+          .collect(Collectors.toList());
+
+    } catch (final IOException e) {
+      throw new TasklistRuntimeException(
+          String.format(
+              "Error removing job worker variable data for flowNodeInstanceId [%s]", varName),
```
Comment (houssain-barouni) [inline tasklist/els-schema/src/main/java/io/camunda/tasklist/store/elasticsearch/ListViewStoreElasticSearch.java:115]: The Exception message needs to be updated

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/VariableService.java [hunk 4]
```diff
@@ -175,17 +195,30 @@ public void persistTaskVariables(
...
+          VariableListViewEntity.createFrom(
+              task.getTenantId(),
+              task.getFlowNodeInstanceId() + "-" + var.getName(),
+              var.getName(),
+              var.getValue(),
+              task.getFlowNodeInstanceId(),
+              tasklistProperties.getImporter().getVariableSizeThreshold(),
+              new ListViewJoinRelation("taskVariable", taskFlowNodeInstanceId)));
     }
+    listViewStore.persistTaskVariables(finalVariablesListViewMap.values());
```
Comment (houssain-barouni) [inline tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/VariableService.java:211]: Does this apply for variables of tasks not migrated to ListView?
We may persist the variable in ListView and not the task itself
Or is the non-migrated task persisted in ListView by the importer?

File: tasklist/webapp/src/main/resources/custom.css
Comment (houssain-barouni) [inline tasklist/webapp/src/main/resources/custom.css:1]: unrelated changes

**Review by houssain-barouni — CHANGES_REQUESTED** [review]
Some comments need to be addressed like scrolling is required to fetch all variables (in ES and OS)

File: tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/VariableService.java (hunk 4 above)
Comment (nathansandi) [reply tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/service/VariableService.java:211]: Does this apply for variables of tasks not migrated to ListView? 
R: At the end it will persist only the variables that will become process variables, once we will remove the task variables from job workers
We are not persisting the Task, only the Variables form the task. But the Process Variables are still necessary to be there.

Or is the non-migrated task persisted in ListView by the importer?
The importer will bring all variables (once it is not possible identify variables coming from V1 or V2). But after task competition, only input variables from Job workers will be removed. 
Non-Migrated tasks wont be imported (Job Workers)

File: tasklist/common/src/main/java/io/camunda/tasklist/entities/listview/ListViewJoinRelation.java [hunk 5]
```diff
@@ -23,6 +23,11 @@ public ListViewJoinRelation(final String name) {
     this.name = name;
   }
 
+  public ListViewJoinRelation(final String name, final String parent) {
+    this.name = name;
+    this.parent = Long.valueOf(parent);
```
[the remark this answers is not in the document]
Comment (nathansandi) [reply tasklist/common/src/main/java/io/camunda/tasklist/entities/listview/ListViewJoinRelation.java]: The nullity check wont be no longer necessary, for this situation, because this constructor just will be called when parent exists (flowNode/scopeKey) 

For the cases that parent is null (process), we have another constructor
```
  public ListViewJoinRelation(final String name) {
    this.name = name;
  }
```

**Review by houssain-barouni — COMMENTED** [review]
Scrolling is still missing in `getVariablesByVariableName `

File: tasklist/els-schema/src/main/java/io/camunda/tasklist/store/opensearch/ListViewStoreOpenSearch.java [hunk 6]
```diff
@@ -118,19 +118,27 @@ public List<VariableListViewEntity> getVariablesByVariableName(final String varN
                       q.term(
                           t ->
                               t.field(TasklistListViewTemplate.VARIABLE_NAME)
-                                  .value(FieldValue.of(varName))));
-
-      final SearchResponse<VariableListViewEntity> searchResponse =
-          osClient.search(searchRequest.build(), VariableListViewEntity.class);
+                                  .value(FieldValue.of(varName))))
+              .size(1000); // Set batch size (1000 is an example, adjust as necessary)
```
Comment (houssain-barouni) [inline tasklist/els-schema/src/main/java/io/camunda/tasklist/store/opensearch/ListViewStoreOpenSearch.java:122]: @nathansandi should we keep it ?

Comment (nathansandi) [reply tasklist/els-schema/src/main/java/io/camunda/tasklist/store/opensearch/ListViewStoreOpenSearch.java]: No :) I left over from my code snippet.

**Review by houssain-barouni — APPROVED** [review]
